coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/HiveToTrinoConverterTest.java

```diff
@@ -106,20 +106,28 @@ public void testViews(String database, String view, String expectedSql) {
             + "FROM \"test\".\"table_with_string_array\" AS \"$cor1\"\n"
             + "CROSS JOIN LATERAL (SELECT \"c\"\nFROM UNNEST(\"if\"(\"$cor1\".\"b\" IS NOT NULL AND CARDINALITY(\"$cor1\".\"b\") > 0, \"$cor1\".\"b\", ARRAY[NULL])) AS \"t0\" (\"c\")) AS \"t1\"" },
 
+        { "test", "view_with_explode_struct_array", "SELECT \"$cor4\".\"a\" AS \"a\", \"t1\".\"c\" AS \"c\"\n"
+            + "FROM \"test\".\"table_with_struct_array\" AS \"$cor4\"\n"
+            + "CROSS JOIN LATERAL (SELECT \"c\"\nFROM UNNEST(TRANSFORM(\"$cor4\".\"b\", x -> ROW(x))) AS \"t0\" (\"c\")) AS \"t1\"" },
+
+        { "test", "view_with_outer_explode_struct_array", "SELECT \"$cor5\".\"a\" AS \"a\", \"t1\".\"c\" AS \"c\"\n"
+            + "FROM \"test\".\"table_with_struct_array\" AS \"$cor5\"\n"
+            + "CROSS JOIN LATERAL (SELECT \"c\"\nFROM UNNEST(TRANSFORM(\"if\"(\"$cor5\".\"b\" IS NOT NULL AND CARDINALITY(\"$cor5\".\"b\") > 0, \"$cor5\".\"b\", ARRAY[NULL]), x -> ROW(x))) AS \"t0\" (\"c\")) AS \"t1\"" },
+
         { "test", "current_date_and_timestamp_view", "SELECT CURRENT_TIMESTAMP, TRIM(CAST(CURRENT_TIMESTAMP AS VARCHAR(65535))) AS \"ct\", CURRENT_DATE, CURRENT_DATE AS \"cd\", \"a\"\nFROM \"test\".\"tablea\"" },
 
-        { "test", "lateral_view_json_tuple_view", "SELECT \"$cor4\".\"a\" AS \"a\", \"t0\".\"d\" AS \"d\", \"t0\".\"e\" AS \"e\", \"t0\".\"f\" AS \"f\"\n"
-            + "FROM \"test\".\"tablea\" AS \"$cor4\"\nCROSS JOIN LATERAL (SELECT "
-            + "\"if\"(\"REGEXP_LIKE\"('trino', '^[^\\\"]*$'), CAST(\"json_extract\"(\"$cor4\".\"b\".\"b1\", '$[\"' || 'trino' || '\"]') AS VARCHAR(65535)), NULL) AS \"d\", "
-            + "\"if\"(\"REGEXP_LIKE\"('always', '^[^\\\"]*$'), CAST(\"json_extract\"(\"$cor4\".\"b\".\"b1\", '$[\"' || 'always' || '\"]') AS VARCHAR(65535)), NULL) AS \"e\", "
-            + "\"if\"(\"REGEXP_LIKE\"('rocks', '^[^\\\"]*$'), CAST(\"json_extract\"(\"$cor4\".\"b\".\"b1\", '$[\"' || 'rocks' || '\"]') AS VARCHAR(65535)), NULL) AS \"f\"\n"
+        { "test", "lateral_view_json_tuple_view", "SELECT \"$cor8\".\"a\" AS \"a\", \"t0\".\"d\" AS \"d\", \"t0\".\"e\" AS \"e\", \"t0\".\"f\" AS \"f\"\n"
```

Comment:
we should reset `$cor` counter between tests

(no change requsted here)

Reply:
Added a commit

Reply:
@losipiuk i suggest separate PR, since we squash-merge here.